src/server/users/models.js

```diff
@@ -179,46 +178,87 @@ export class CreatedPostManager {
  *    state: 'alive',
  *    key: 'item-8523910540005-dd3860f5-b82e-473b-1234-ead0f190b000',
  *    userKey: 'user-8000000000000-uuiduuid-uuid-uuid-uuid-uuiduuiduuid',
- *    entity: 'image',
- *    imageUrl: 'url-of-image-8523569761934-dd3860f5-b82e-473b-1234-ead0f190b000'
+ *    isCreatedByUser: false,
+ *    images: [
+ *              {
+ *                imageKey: 'image-8523569763000-dd3860f5-b82e-473b-1234-ead0f190b000',
+ *                imageUrl: 'url-of-image-8523569763000-dd3860f5-b82e-473b-1234-ead0f190b000'
+ *              }      
+ *            ]
  * }
 **/
-  static getPost({entity, itemKey, imageKey}, cb) {
-    db.get(itemKey, (err, item) => {
-      if (err) {
-        return cb(err);
+  static fetchPost({isCreatedByUser, itemKey, imageKeys}) {
+    return getPromise(itemKey).then((item)=>{
+      item.isCreatedByUser = isCreatedByUser;
+      item.images = ImageManager.getImageObjList(imageKeys);
+      return item;
+    });
+  }
+/**
+ * combine key sets for unique item key
+ * @param {Array} posts example:
+ * [
+ *    {  entity  : 'image',
+ *       itemKey : 'item-8523910540005-dd3860f5-b82e-473b-1234-ead0f190b000',
+ *       imageKey: 'image-8523569761934-dd3860f5-b82e-473b-1234-ead0f190b000'
+ *    }
+ *    {  entity  : 'item',
+ *       itemKey : 'item-8523910540005-dd3860f5-b82e-473b-1234-ead0f190b000',
+ *       imageKey: 'image-8523569761934-dd3860f5-b82e-473b-1234-ead0f190b999'
+ *    }
+ * ]
+ * @return {Array} item example:
+ * [
+ *    {  isCreatedByUser : true,
+ *       itemKey : 'item-8523910540005-dd3860f5-b82e-473b-1234-ead0f190b000',
+ *       imageKeys :
+ *        [
+ *          'image-8523569761934-dd3860f5-b82e-473b-1234-ead0f190b000',
+ *          'image-8523569761934-dd3860f5-b82e-473b-1234-ead0f190b999'
+ *        ]
+ *    }
+ * ]
+ */
+  static combinePosts(posts) {
+    const objMap = new Map();
+    const ret = [];
+    posts.map((post)=> {
+      if (!objMap.get(post.itemKey)) {
+        // if there's no value of itemKey, init an object.
+        objMap.set(post.itemKey, {
+          isCreatedByUser: 'false',
+          itemKey: post.itemKey,
+          imageKeys: []
+        });
       }
-      item.entity = entity;
-      item.imageUrl = new S3Connector().getImageUrl(imageKey);
-      return cb(null, item);
+      // if this item is created by this user -> isCreatedByUser = true
+      objMap.get(post.itemKey).isCreatedByUser =
+        (post.entity === ENTITY.ITEM) || objMap.get(post.itemKey).isCreatedByUser;
+      objMap.get(post.itemKey).imageKeys.push(post.imageKey);
     });
+    // convert map to array for return
+    for (const value of objMap.values()) {
```

Comment:
check this out!  you can just code like return Array.from(objMap.values());
[https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/from](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/from)